### ISSUE TITLE
OpenID Connect Provider does not pre-populate user email even if provided

### ISSUE THREAD

**Issue by david-code** [pennersr/django-allauth#3257]
# Description

Hello, thank you for this very useful library! I'd like to outline a problem I ran into when integrating an OpenID Connect Provider.

The problem arises when the Django application using `OpenIDConnectProvider` queries the [UserInfo endpoint](https://openid.net/specs/openid-connect-core-1_0.html#UserInfo) on the other server to complete the signup process. It might return a payload like the following:

```json
{
        "sub": "60e1123a-3583-4f04-a0ef-1037e97c2276",
        "email": "test@example.org",
        "given_name": "Test",
        "family_name": "User"
}
```

However, this email will not be used to properly set the user's email in the adapters `populate_user` method. The error seems to be from the following code:

```python
         email = data.get("email")
         user_email(user, valid_email_or_none(email) or "")
```

The problem is that the `data` above is not overriden by `OpenIDConnectProvider` and thus returned by `Provider.extract_common_fields`. This base clase method returns an empty dictionary. Therefore despite the fact we have a valid email attached to the `sociallogin`, the email will always be an empty string.

An easy solution would be to override `extract_common_fields` in `OpenIDConnectProvider` to output the email key. Even if the email scope is not included, it should be fine to return an empty dictionary (it will work the same as it does right now). However, I noticed that in the `Provider.sociallogin_from_response` we already have gotten a list of email addresses from the response. The `sociallogin` holding the email addresses is also available in the `populate_user` method. Would a more general solution be to change the code above to

```python
         email = data.get("email") or (sociallogin.emal_addresses and sociallogin.email_adresses[0])
         user_email(user, valid_email_or_none(email) or **"")**
```

I would like to implement the first solution regardless (unless you disagree and think it should be done in a different way). What do you think about the second solution proposed? It would affect all of the providers and would probably need more testing.

**Comment by stumpylog** [comment]
Sounds similar or the same as to #3241

**Comment by david-code** [comment]
@stumpylog you're right, it looks like the same issue.

**Comment by pennersr** [comment]
Closed via e1acb766f5bc578009ceb4db414f5730ce02adab

**Comment by pennersr** [comment]
https://codeberg.org/allauth/django-allauth/issues/3257
